### ISSUE TITLE
[SCM-976] GitExe changelog does not work in if the user has defined a custom format

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-scm#1205]
**[Niels Basjes](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nielsbasjes)** opened **[SCM-976](https://issues.apache.org/jira/browse/SCM-976?redirect=false)** and commented

Since a long time the headers for the output of "git whatchanged" have been changed, as a consequence there is no more output from this command.
The reason is that the output of this command used to be something like this:

```
commit 2d22f2da71ef279c4759b1c09cdb8282b63f0cf6
Author: Mark Struberg <struberg@yahoo.de>
Date:   2007-11-24 01:10:06 +0100

```

yet currently it looks like this:

```
commit 4fc34cfa14f2e72506187b03a492ce55ed459d4c (HEAD -> master, origin/master, origin/HEAD)
Author:     Michael Osipov <michaelo@apache.org>
AuthorDate: 2022-01-08 21:02:12 +0100
Commit:     Michael Osipov <michaelo@apache.org>
CommitDate: 2022-01-08 21:02:12 +0100

```

---

**Affects:** 2.0.0-M1

**Remote Links:**
- [GitHub Pull Request #134
](https://github.com/apache/maven-scm/pull/134)

0 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas)** commented

Just a few thoughts:
* git whatchanged is kept only for "historical" reasons. From man: "New users are encouraged to use git-log instead. The whatchanged command is essentially the same as git-log but defaults to show the raw format diff output and to skip merges. The command is kept primarily for historical reasons..."
* am using git 2.25.1 on my linux box and my output is like the first "used to be" output (but adds HEAD pointers like second one). Am unsure what may drive the output, it may be my env, but am unsure you can do it in "generic" way.
* why not switch to git-log instead?

My example output:

```
commit 05b67a4f72f3f64306637ccdf0d2f800616dee5c (HEAD -> master, origin/master, origin/HEAD)
Author: Tamas Cservenak <tamas@cservenak.net>
Date:   Mon Feb 7 19:33:01 2022 +0100

    [MINDEXER-104] Migrate to Lucene 8.x (#169)
    
    * Eliminate use of deprecated Lucene API
...       
```

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[Niels Basjes](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nielsbasjes), please change the issue description. I don't consider it to be correct anymore.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [178b111bb9976caafd97bce174260f12f0faa8d6](https://gitbox.apache.org/repos/asf?p=maven-scm.git;a=commit;h=178b111bb9976caafd97bce174260f12f0faa8d6) and with [0f8b37580a08f3237b3939073ab49fb8c1070aa5](https://gitbox.apache.org/repos/asf?p=maven-scm.git;a=commit;h=0f8b37580a08f3237b3939073ab49fb8c1070aa5) for `maven-scm-1.x` branch.
